library/swift/HeadersBuilder.swift

```diff
@@ -3,14 +3,19 @@ import Foundation
 private let kRestrictedPrefixes = [":", "x-envoy-mobile"]
 
 private func isRestrictedHeader(name: String) -> Bool {
-  return name == "host" || kRestrictedPrefixes.contains { name.hasPrefix($0) }
+  let isHostHeader = name.caseInsensitiveCompare("host") == .orderedSame
+  let hasRestrictedPrefix = kRestrictedPrefixes
```

Comment:
Avoid eagerly computing this if the name is `host`

```suggestion
  lazy var hasRestrictedPrefix = kRestrictedPrefixes
```